### PR TITLE
Update project config to support linux builds

### PR DESCRIPTION
**Pull request by StCyr**
This is a config update proposal to be able to build the project under linux (and develop with a development build directly, without simulator). Tested, it works here.

Fixes https://github.com/mozzius/graysky/issues/131

# Breaking changes

* The script to build the app for the ios simulator has been renamed `build:dev-client:ios-simulator` (the `build:dev-client:ios` script now builds a development build using the EAS build service)

# Reviewing tips / possible improvements

* Should I really commit the .env file? shouldn't it be better to document how to create this file in the README.md file?
* Should I really set the `SIMULATOR` environment variable in the .env file? Wouldn't it be better to define it in the eas.json file?
* Can I do something nicer than renaming the script to build the app for the ios simulator?

### CODE REVIEW

**Comment by mozzius** [comment]
Hi, thanks for the contribution!

No need to mess with .env files, the SIMULATOR environment variable is already in eas.json. Is there a misleading comment I've left somewhere by mistake?

File: .env
Comment (mozzius) [inline .env:1]: No need to commit this

File: apps/expo/app.config.ts [hunk 1]
```diff
@@ -47,7 +46,7 @@ const defineConfig = (_: ConfigContext): ExpoConfig => ({
     },
   },
   android: {
-    package: "dev.mozzius.graysky",
+    package: process.env.APP_ID,
```
Comment (mozzius) [inline apps/expo/app.config.ts:49]: Could you default it to `dev.mozzius.graysky` if a value for APP_ID is unspecified?

File: apps/expo/package.json [hunk 2]
```diff
@@ -126,7 +126,8 @@
     "eas-build-pre-install": "./scripts/pre-install.sh",
     "build:ios": "./scripts/build-ios.sh",
     "build:android": "./scripts/build-android.sh",
-    "build:dev-client:ios": "eas build --platform ios --profile development-simulator --non-interactive --local --output=\"./graysky.tar.gz\" --wait && eas build:run -p ios --path=\"graysky.tar.gz\"",
+    "build:dev-client:ios": "eas build --platform ios --profile development",
```
Comment (mozzius) [inline apps/expo/package.json:129]: Why not run it locally? Is this a linux limitation? Bare in mind it costs $2 per run

**Review by mozzius — CHANGES_REQUESTED** [review]
Could you revert the changes with regards to the .env file? You should specify the APP_ID via EAS secrets rather than commiting the .env file

File: apps/expo/package.json (hunk 2 above)
Comment (StCyr) [reply apps/expo/package.json:129]: Yes, the simulator doesn't run on linux afaik.

The EAS build service is free for hobbyists for up to 15 iOS build per month

Comment (mozzius) [reply apps/expo/package.json:129]: Ok fair enough

Comment (StCyr) [reply apps/expo/package.json:129]: (local builds don't work neither)

File: apps/expo/app.config.ts (hunk 1 above)
Comment (StCyr) [reply apps/expo/app.config.ts]: ok, done. And for ios:bundleIdentifier too

File: .env
Comment (StCyr) [reply .env:1]: You've explained how to set APP_ID via EAS secrets, that's fine. But, there's still an issue with EAS_PROJECT_ID: How I am supposed to set it? Also via EAS secrets?

And there's also an issue with SIMULATOR: It's set in eas.json but not for the development profile:

```
  "build": {
    "development-simulator": {
      "developmentClient": true,
      "ios": {
        "simulator": true
      },
      "env": {
        "SIMULATOR": "true"
      }
    },
    "development": {
      "developmentClient": true,
      "distribution": "internal",
      "ios": {
        "resourceClass": "m-medium"
      }
    },
```

and I'm not sure adding it won't impact android builds... Maybe, I should split this "development" profile in 2: "development-ios" and "development-android".... What do you think?

**Comment by StCyr** [comment]
Didn't you merge the PR too early @mozzius ? the .env file is still in there, and I've still some issue to remove it. see https://github.com/mozzius/graysky/pull/132#discussion_r1380487557

File: .env
Comment (mozzius) [reply .env:1]: Set EAS_PROJECT_ID in a .env file - you don't need to commit it. It's just how your local eas-cli links the project to EAS.

I've not tested if SIMULATOR: true is needed for non-simulator builds. I'll add it anyway, why not

Comment (StCyr) [reply .env:1]: For iOS, SIMULATOR:true is needed to get UIViewControllerBasedStatusBarAppearance to false (which is needed for the development client to run on my iphone 11):

see app.config.ts

```
    infoPlist: {
      // should be true but simulator builds don't like this
      UIViewControllerBasedStatusBarAppearance:
        process.env.SIMULATOR === "true" ? false : true,
```

Comment (StCyr) [reply .env:1]: shall I create a new PR to remove the .env file (and reset the corresponding .gitignore rule)?

**Pull request merged**